### DISCUSSION TITLE
Complications in Setting Up the Project

### DISCUSSION THREAD

**Discussion by TheCharismaticEnigma10** [narayan954/dummygram#129] [General]
@kshitij01042002 
Hey man. I'm trying to run this project on my local machine but I'm not able to break through. I tried to use git bash
to open it but I'm not getting anything. I read the README file and followed the steps but got nowhere. Is there any way that I can run this on my local server? I'd be glad if you could help me out.

**Comment by narayan954** [comment]
Can you share some pictures of issue / elaborate your issue

**Comment by kshitij01042002** [comment]
@TheCharismaticEnigma share the the snapshots of the issues you are facing while setting up the environment

**Comment by TheCharismaticEnigma10** [comment]
@kshitij01042002  @narayan954 

I could see nothing when I tried to open it from VS code just a little toggle button at the end. But when I use the npm live server I can see the entire app. 
How can I get the HTML code? I'm quite new to all of this

**Comment by TheCharismaticEnigma10** [comment]


**Reply by narayan954** [reply]
by using command `npm install -g n` you are asking npm to install "n" package which gives this obvious error. 
Try using command without "n" like `npm install` or `npm i` , doing this will command npm to install everything mentioned in package.json file located in the directory.

**Reply by narayan954** [reply]
> > by using command `npm install -g n` you are asking npm to install "n" package which gives this obvious error. Try using command without "n" like `npm install` or `npm i` , doing this will command npm to install everything mentioned in package.json file located in the directory.
> 
> Also `-g` flag is used to install the package globally, which is not necessary in this case so you can avoid using this flag

**Reply by narayan954** [reply]
Also, in the error message you can read that "n" package's requirement is !win32 os or not win32 os and your system specification is win32, so its failing

**Reply by TheCharismaticEnigma10** [reply]
> by using command `npm install -g n` you are asking npm to install "n" package which gives this obvious error. Try using command without "n" like `npm install` or `npm i` , doing this will command npm to install everything mentioned in package.json file located in the directory.

Thank You mate.  Can you please recommend some NODE.js concepts/topics that I should learn to understand these basic concepts and terms?  @narayan954

**Reply by narayan954** [reply]
well Node js is not needed/used at all in this project.
but you can get to know about npm which just provides these package/modules.

**Reply by TheCharismaticEnigma10** [reply]
@narayan954  Hey mate. I can now see the project but when I use the command npm install, I get all sorts of errors. Should I use the --force option? I've read that you shouldn't use the force option. 
![Screenshot (4)](https://user-images.githubusercontent.com/102666605/208405081-d73c2a9b-94dc-40d6-86b1-d9041686bd20.png)
![Screenshot (5)](https://user-images.githubusercontent.com/102666605/208405087-89fb48e0-d61f-4602-91aa-8018bef20679.png)

**Reply by narayan954** [reply]
> @narayan954 Hey mate. I can now see the project but when I use the command npm install, I get all sorts of errors. Should I use the --force option? I've read that you shouldn't use the force option. ![Screenshot (4)](https://user-images.githubusercontent.com/102666605/208405081-d73c2a9b-94dc-40d6-86b1-d9041686bd20.png) ![Screenshot (5)](https://user-images.githubusercontent.com/102666605/208405087-89fb48e0-d61f-4602-91aa-8018bef20679.png)

you can try force command to install the dependencies as according to this error there're some dependency conflicts. You can google the error to know more

**Comment by TheCharismaticEnigma10** [comment]
@narayan954  And would you please tell me how can I see the HTML CODE?  
![Screenshot (6)](https://user-images.githubusercontent.com/102666605/208405719-455107be-47aa-4d49-92d6-39b18ee6d457.png)

**Reply by narayan954** [reply]
This is a react project, so there's no exclusive html code, there's only a blank html page as you can see from this code and JavaScript in other files is manipulating the DOM.

**Reply by TheCharismaticEnigma10** [reply]
So is there no way for me to make some changes to the project with my current skill set? @narayan954

**Reply by narayan954** [reply]
> So is there no way for me to make some changes to the project with my current skill set? @narayan954

It'd be hard for you, but I believe you can work on CSS/Design part of the project without knowledge of react, you can try to take a quick basic intro to the file structure of a react project to understand where the things are and can go into changing designs.

**Reply by kshitij01042002** [reply]
@TheCharismaticEnigma follow some youtube tutorials if you want, get idea about how react works and you'll be good to go

**Reply by TheCharismaticEnigma10** [reply]
> > So is there no way for me to make some changes to the project with my current skill set? @narayan954
> 
> It'd be hard for you, but I believe you can work on CSS/Design part of the project without knowledge of react, you can try to take a quick basic intro to the file structure of a react project to understand where the things are and can go into changing designs.

AAHH! It's all good. I'm in the midst of learning advanced JS and I'll come back here once I've upskilled myself as I think that'd be a better route to take. But anyways,  Thanks for the help mates because you've been supportive. I'LL BE BACK!

**Comment by TheCharismaticEnigma10** [comment]
@narayan954  Hey man.  How are you doing? I actually wanted to ask you something. I'm learning advanced JS at the moment. So,
I had a couple of questions and I'd be glad if you'd answer them. 
1. What should I do after learning JS? Should I start learning REACT concepts right away? 
2. Do I need to memorize all the APIs like geolocation API and all of what I'm learning right now? 
3. Do you have a reference to  an article that has the entire list of concepts that I should learn before I can contribute to projects like this ? 

Thanks for answering !

**Reply by narayan954** [reply]
hi @TheCharismaticEnigma I'm doing fine, thanks for asking, and what about you? 
Assuming you know the concepts of html and css as well, you can start react right away. 
No, you dont need to memorize those API, you might learn them OTG (while on a project).
There's no such explicit requirement, you can start right away once you get to understand the file structure. Any playlist on youtube might tell you the concepts you need, like routing, user auth through firebase, firebase storage,etc and you can learn these OTG while on this or any project as well. 
PS: You may start participating in more open-source programs like this as well to get such projects to work on and learn.

Regards

**Reply by TheCharismaticEnigma10** [reply]
> hi @TheCharismaticEnigma I'm doing fine, thanks for asking, and what about you? Assuming you know the concepts of html and css as well, you can start react right away. No, you dont need to memorize those API, you might learn them OTG (while on a project). There's no such explicit requirement, you can start right away once you get to understand the file structure. Any playlist on youtube might tell you the concepts you need, like routing, user auth through firebase, firebase storage,etc and you can learn these OTG while on this or any project as well. PS: You may start participating in more open-source programs like this as well to get such projects to work on and learn.
> 
> Regards

@narayan954 All good in the hood brother.  Are there any open-source projects solely using vanilla JS? I only get the regular To-Do list projects and all of that.  Appreciate the help once again !
